graspy/match/faq.py

```diff
@@ -0,0 +1,168 @@
+        "barycenter" : the non-informative “flat doubly stochastic matrix,” J=1*1^T/n, i.e the barycenter of
+        the feasible region
+        "rand" : some random point near J, (J+K)/2, where K is some random doubly stochastic matrix
+        
+    Attributes
+    ----------
+    
+    perm_inds_ : array, size (n,1) where n is the number of vertices in the graphs fitted
+        The indices of the optimal permutation found via FAQ
+        
+    ofv_ : float
+        The objective function value of for the optimal permutation found
+        
+        
+    References
+    ----------
+    .. [1] J. T. Vogelstein, J. M. Conroy, V. Lyzinski, L. J. Podrazik, S. G. Kratzer, E. T. Harley, 
+        D. E. Fishkind, R. J. Vogelstein, and C. E. Priebe, “Fast approximate quadratic programming 
+        for graph matching,” PLOS one, vol. 10, no. 4, p. e0121002, 2015.
+
+    """
+
+    def __init__(self, n_init, init_method):
+
+        if n_init > 0 and type(n_init) is int:
+            self.n_init = n_init
+        else:
+            msg = '"n_init" must be a positive integer'
+            raise TypeError(msg)
+        if init_method == "rand":
+            self.init_method = "rand"
+        elif init_method == "barycenter":
+            self.init_method = "barycenter"
+            n_init = 1
+        else:
+            msg = 'Invalid "init_method" parameter string'
+            raise ValueError(msg)
+
+    def fit(self, A, B):
+        """
+        Fits the model with two assigned adjacency matrices
+        
+        Parameters
+        ---------
+        A : 2d-array, square
+            A square adjacency matrix
+            
+        B : 2d-array, square
+            A square adjacency matrix
+        
+        Returns
+        -------
+        
+        self : returns an instance of self
+        """
+        if A.shape[0] != B.shape[0]:
+            msg = "Matrix entries must be of equal size"
+            raise ValueError(msg)
+        elif A.shape[0] != A.shape[1] or B.shape[0] != B.shape[1]:
+            msg = "Matrix entries must be square"
+            raise ValueError(msg)
+        elif (A >= 0).all() == False or (B >= 0).all() == False:
+            msg = "Matrix entries must be greater than or equal to zero"
+            raise ValueError(msg)
+        else:
+
+            n = A.shape[0]  # number of vertices in graphs
+            At = np.transpose(A)  # A transpose
+            Bt = np.transpose(B)  # B transpose
+            score = math.inf
+            perm_inds = np.zeros(n)
+
+            for i in range(self.n_init):
+
+                # setting initialization matrix
+                if self.init_method == "rand":
+                    sk = SinkhornKnopp()
+                    K = np.random.rand(
+                        n, n
+                    )  # generate a nxn matrix where each entry is a random integer [0,1]
+                    for i in range(10):  # perform 10 iterations of Sinkhorn balancing
```

Comment:
is there a reason to ever make this a parameter rather than just 10?

Reply:
I don't think so. I do this because jovo does in his paper. I think a similar result would be achieved if I called the function once